bi_sql_editor/models/bi_sql_view.py

```diff
@@ -730,9 +730,12 @@ def _refresh_size(self):
     def check_manual_fields(self, model):
         # check the fields we need are defined on self, to stop it going
         # early on install / startup - particularly problematic during upgrade
-        if "group_operator" in table_columns(
-            self.env.cr, "bi_sql_view_field"
-        ) and model._name.startswith(self._model_prefix):
+        len_model_prefix = len(self._model_prefix)
+        if len(model._name) < len_model_prefix:
+            return
+        if model._name[:len_model_prefix] != self._model_prefix:
+            return
+        if "group_operator" in table_columns(self.env.cr, "bi_sql_view_field"):
```

Comment:
@paulius-sladkevicius I would suspect that the slow part would be the `table_columns()` call (that would explain the >2x query increase during module loading/testing), not the `startswith()` check. In such case, switching the operands should be enough, as then `and` would [short-circuit](https://docs.python.org/3/tutorial/datastructures.html?highlight=short%20circuit#more-on-conditions). IIUC that would be enough as the model prefix would not match in most cases.
```suggestion
        if model._name.startswith(self._model_prefix) and "group_operator" in table_columns(self.env.cr, "bi_sql_view_field"):
```

Reply:
You're totally right, many thanks!